src/game.cpp

```diff
@@ -6792,6 +6792,18 @@ Creature const* game::critter_at( const tripoint &p ) const
     return const_cast<game*>(this)->critter_at( p );
 }
 
+monster * game::summon_mon( const std::string id, const tripoint &p )
+{
+    monster mon(GetMType(id));
+    // Set their last upgrade check to the current day.
+    mon.reset_last_load();
+    mon.spawn(p);
+    add_zombie(mon);
+
+    // Return the non-local copy of the monster
+    return &zombie(critter_tracker.mon_at(p));
```

Comment:
You could make `add_zombie` return the new monster, than `return add_zombie(...)`. `add_zombie` already returns a bool, but is only used in catalua.cpp

But you want to check the result of `add_zombie` anyway, there are several situations when it *wont* have add a zombie, calling `critter_tracker.mon_at(p)` will fail then:
- the monster id was mon_null,
- the monsters location is outside the map
- (most likely) the monster is blacklisted